### PR TITLE
Add Amazon Cognito SSO support to iDempiere #252

### PR DESCRIPTION
**Pull request by etantg**
Documentation: https://docs.google.com/document/d/1p5PCtAij18L-PpoMu-aT5WEdnQb1jTowwgHNkfAMmWU/edit?usp=sharing

### CODE REVIEW

File: com.trekglobal.idempiere.rest.api/src/com/trekglobal/idempiere/rest/api/model/MOIDCService.java [hunk 1]
```diff
@@ -89,6 +90,9 @@ public class MOIDCService extends X_REST_OIDCService implements ImmutablePOSuppo
 	/** HTTP header for AD_Language */
 	public static final String LANGUAGE_HEADER = "X-ID-Language";
 	
+	/** HTTP header for AD_Org.Value. */ 
+	public static final String IDTOKEN_HEADER = "IdToken";
```
Comment (hengsin) [inline com.trekglobal.idempiere.rest.api/src/com/trekglobal/idempiere/rest/api/model/MOIDCService.java:94]: Should use X- prefix for custom http header. My suggestion is to use X-ID-IdToken (X for custom and ID for iDempiere).

Comment (etantg) [reply com.trekglobal.idempiere.rest.api/src/com/trekglobal/idempiere/rest/api/model/MOIDCService.java]: Changed to X-ID-IdToken

**Comment by etantg** [comment]
- Changed Cognito group to map AD_Role.Name without space
- Added README.txt and Testing Steps.md

**Review by hengsin — APPROVED** [review]
looks good

**Comment by CarlosRuiz-globalqss** [comment]
@etantg - I think the Testing Steps.md is not showing the images.

Maybe it can be better to point it to a wiki page or public google doc?

**Comment by etantg** [comment]
@CarlosRuiz-globalqss - Created a wiki page: https://wiki.idempiere.org/en/OIDC_SSO_Amazon_Cognito_Testing_Steps. Removed the Testing Steps.md, replaced with Testing Steps.txt.

**Pull request merged**